From 06d05149e8f24aaf2acc9ff86e134b2ee5c129eb Mon Sep 17 00:00:00 2001
From: Ye Li <ye.li@nxp.com>
Date: Mon, 2 Mar 2026 16:30:00 +0800
Subject: [PATCH 10/24] iMX94: Add XSPI FCB copy image

Add FCB copy image to OEM container for all XSPI boot targets
It is needed for XSPI boot on OEM closed LC.

Upstream-Status: Pending

Signed-off-by: Ye Li <ye.li@nxp.com>
---
 iMX94/soc.mak | 36 ++++++++++++++++++------------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/iMX94/soc.mak b/iMX94/soc.mak
index aaa037a..257ed7b 100644
--- a/iMX94/soc.mak
+++ b/iMX94/soc.mak
@@ -332,14 +332,14 @@ flash_a55: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) u-boot-atf-container.img $(SPL_A55_IM
 flash_a55_xspi: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) fcb.bin u-boot-atf-container.img $(SPL_A55_IMG) $(OEI_IMG_M33) $(OEI_M33_DDR_IMG)
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(XSPI_FAST_HASH) -append $(AHAB_IMG) -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   $(V2X_DUMMY) -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) -out flash.bin
+		   $(V2X_DUMMY) -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 	$(call append_container,u-boot-atf-container.img,1)
 	$(call append_fcb)
 
 flash_a55_xspi_crrm: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) fcb.bin crrm-container.img $(SPL_A55_IMG) $(OEI_IMG_M33) $(OEI_M33_DDR_IMG)
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(XSPI_FAST_HASH) -u 1 -append $(AHAB_IMG) -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   $(V2X_DUMMY) -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) -out flash.bin
+		   $(V2X_DUMMY) -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 	$(call append_container,crrm-container.img,1)
 	$(call append_fcb)
 
@@ -349,7 +349,7 @@ flash_a55_xspi_oem_fastboot: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(SPL_A55_IMG) $(OE
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(XSPI_FAST_HASH) -images_hash sha256 \
 		   -append $(AHAB_IMG) -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   $(V2X_DUMMY) -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) -out flash.bin
+		   $(V2X_DUMMY) -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 	$(call append_container,u-boot-atf-container.img,1)
 	$(call append_fcb)
 
@@ -376,7 +376,7 @@ flash_a55_m70_ddr: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) u-boot-atf-container.img $(SP
 flash_a55_no_ahabfw_flexspi: $(MKIMG) $(MCU_IMG) $(SPL_A55_IMG) $(OEI_IMG_M33) fcb.bin u-boot-atf-container.img
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(CNTR_FLAG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   $(V2X_DUMMY) -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) -out flash.bin
+		   $(V2X_DUMMY) -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 	$(call append_container,u-boot-atf-container.img,1)
 	$(call append_fcb)
 
@@ -392,7 +392,7 @@ flash_m70: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33)
 flash_m70_xspi: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33) fcb.bin
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(XSPI_FAST_HASH) -append $(AHAB_IMG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   -m7 $(M70_IMG) 0 $(M70_TCM_ADDR) $(M70_TCM_ADDR_ALIAS) -out flash.bin
+		   -m7 $(M70_IMG) 0 $(M70_TCM_ADDR) $(M70_TCM_ADDR_ALIAS) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 flash_m70_ddr: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33)
@@ -403,7 +403,7 @@ flash_m70_ddr: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33)
 flash_m70_ddr_xspi: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33) fcb.bin
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(XSPI_FAST_HASH) -append $(AHAB_IMG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   -m7 $(M70_IMG) 0 $(M70_DDR_ADDR) $(M70_DDR_ADDR) -out flash.bin
+		   -m7 $(M70_IMG) 0 $(M70_DDR_ADDR) $(M70_DDR_ADDR) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 flash_m70_no_ahabfw: $(MKIMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33)
@@ -414,7 +414,7 @@ flash_m70_no_ahabfw: $(MKIMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33)
 flash_m70_no_ahabfw_xspi: $(MKIMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33) fcb.bin
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(CNTR_FLAG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   -m7 $(M70_IMG) 0 $(M70_TCM_ADDR) $(M70_TCM_ADDR_ALIAS) -out flash.bin
+		   -m7 $(M70_IMG) 0 $(M70_TCM_ADDR) $(M70_TCM_ADDR_ALIAS) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 flash_m70_ddr_no_ahabfw: $(MKIMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33)
@@ -425,7 +425,7 @@ flash_m70_ddr_no_ahabfw: $(MKIMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33)
 flash_m70_ddr_no_ahabfw_xspi: $(MKIMG) $(MCU_IMG) $(M70_IMG) $(OEI_IMG_M33) fcb.bin
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(CNTR_FLAG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   -m7 $(M70_IMG) 0 $(M70_DDR_ADDR) $(M70_DDR_ADDR) -out flash.bin
+		   -m7 $(M70_IMG) 0 $(M70_DDR_ADDR) $(M70_DDR_ADDR) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 ## AHAB_IMG shall include both ELE and V2X containers ##
@@ -433,7 +433,7 @@ flash_m70_ddr_xspi_oem_fastboot: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M70_IMG) $(OE
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(XSPI_FAST_HASH) -images_hash sha256 -dev flexspi \
 		   -append $(AHAB_IMG) -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   -m7 $(M70_IMG) 0 $(M70_DDR_ADDR) $(M70_DDR_ADDR) $(V2X_DUMMY) -out flash.bin
+		   -m7 $(M70_IMG) 0 $(M70_DDR_ADDR) $(M70_DDR_ADDR) $(V2X_DUMMY) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 flash_m71: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M71_IMG) $(OEI_IMG_M33)
@@ -444,7 +444,7 @@ flash_m71: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M71_IMG) $(OEI_IMG_M33)
 flash_m71_xspi: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M71_IMG) $(OEI_IMG_M33) fcb.bin
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(XSPI_FAST_HASH) -append $(AHAB_IMG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) -out flash.bin
+		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 flash_m71_no_ahabfw: $(MKIMG) $(MCU_IMG) $(M71_IMG) $(OEI_IMG_M33)
@@ -455,7 +455,7 @@ flash_m71_no_ahabfw: $(MKIMG) $(MCU_IMG) $(M71_IMG) $(OEI_IMG_M33)
 flash_m71_no_ahabfw_xspi: $(MKIMG) $(MCU_IMG) $(M71_IMG) $(OEI_IMG_M33) fcb.bin
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(CNTR_FLAG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) -out flash.bin
+		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 flash_m70_m71: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M70_IMG) $(M71_IMG) $(OEI_IMG_M33)
@@ -468,7 +468,7 @@ flash_m70_m71_xspi: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M70_IMG) $(M71_IMG) $(OEI_
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(XSPI_FAST_HASH) -append $(AHAB_IMG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
 		   -m7 $(M70_IMG) 0 $(M70_TCM_ADDR) $(M70_TCM_ADDR_ALIAS) \
-		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) -out flash.bin
+		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 
@@ -482,7 +482,7 @@ flash_m70_m71_no_ahabfw_xspi: $(MKIMG) $(MCU_IMG) $(M70_IMG) $(M71_IMG) $(OEI_IM
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(CNTR_FLAG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
 		   -m7 $(M70_IMG) 0 $(M70_TCM_ADDR) $(M70_TCM_ADDR_ALIAS) \
-		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) -out flash.bin
+		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 flash_m33s: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M33S_IMG) $(OEI_IMG_M33)
@@ -495,10 +495,10 @@ flash_m33s_ddr: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M33S_IMG) $(OEI_IMG_M33)
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
 		   -m33 $(M33S_IMG) 1 $(M33S_DDR_ADDR) $(M33S_DDR_ADDR) -out flash.bin
 
-flash_m33s_xspi: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M33S_IMG) $(OEI_IMG_M33)
+flash_m33s_xspi: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M33S_IMG) $(OEI_IMG_M33) fcb.bin
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(XSPI_FAST_HASH) -append $(AHAB_IMG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   -m33 $(M33S_IMG) 1 $(M33S_TCM_ADDR) $(M33S_TCM_ADDR_ALIAS) -out flash.bin
+		   -m33 $(M33S_IMG) 1 $(M33S_TCM_ADDR) $(M33S_TCM_ADDR_ALIAS) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 flash_m33s_no_ahabfw: $(MKIMG) $(MCU_IMG) $(M33S_IMG) $(OEI_IMG_M33)
@@ -506,10 +506,10 @@ flash_m33s_no_ahabfw: $(MKIMG) $(MCU_IMG) $(M33S_IMG) $(OEI_IMG_M33)
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
 		   -m33 $(M33S_IMG) 1 $(M33S_TCM_ADDR) $(M33S_TCM_ADDR_ALIAS) -out flash.bin
 
-flash_m33s_no_ahabfw_xspi: $(MKIMG) $(MCU_IMG) $(M33S_IMG) $(OEI_IMG_M33)
+flash_m33s_no_ahabfw_xspi: $(MKIMG) $(MCU_IMG) $(M33S_IMG) $(OEI_IMG_M33) fcb.bin
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(CNTR_FLAG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
-		   -m33 $(M33S_IMG) 1 $(M33S_TCM_ADDR) $(M33S_TCM_ADDR_ALIAS) -out flash.bin
+		   -m33 $(M33S_IMG) 1 $(M33S_TCM_ADDR) $(M33S_TCM_ADDR_ALIAS) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 flash_m33s_m71_no_ahabfw: $(MKIMG) $(MCU_IMG) $(M33S_IMG) $(M71_IMG) $(OEI_IMG_M33)
@@ -522,7 +522,7 @@ flash_m33s_m71_no_ahabfw_xspi: $(MKIMG) $(MCU_IMG) $(M33S_IMG) $(M71_IMG) $(OEI_
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(CNTR_FLAG) -dev flexspi -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
 		   -m7 $(M33S_IMG) 0 $(M33S_TCM_ADDR) $(M33S_TCM_ADDR_ALIAS) \
-		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) -out flash.bin
+		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 		   $(call append_fcb)
 
 flash_m33s_m70: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M33S_IMG) $(M70_IMG) $(OEI_IMG_M33)
